### PR TITLE
Add qnn batch_matmul operator

### PR DESCRIPTION
**Pull request by elvin-n**
+ added support of the different out type for x86 batch_matmul

### CODE REVIEW

**Review by jwfromm — COMMENTED** [review]
These changes look excellent, I really like how you've minimized code duplication in the shape functions. I'm not quite sure why CI is failing, the error message is saying that Relay expects the output shape to be [16, 32] instead of [16, 16] like its supposed to be. I couldn't find any change in the shape functions that would make relay think that but there must be something subtle.

**Comment by elvin-n** [comment]
@jwfromm @masahi Could you please review?

File: python/tvm/relay/qnn/op/qnn.py [hunk 1]
```diff
@@ -682,6 +682,42 @@ def subtract(
...
+    Computes batch matrix multiplication of `x` and `y` when `x` and `y` are data
+    in batch.
+
+    .. math::
+
+        \mbox{batch_matmul}(x, y)[i, :, :] = \mbox{matmul}(x[i, :, :], y[i, :, :]^T)
+
+    Parameters
+    ----------
+    x : tvm.relay.Expr
```
Comment (masahi) [inline python/tvm/relay/qnn/op/qnn.py:696]: `x, y`

File: tests/python/relay/test_op_qnn_batch_matmul.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    x_shape, y_shape, output_shape = (1, 4, 5), (1, 3, 5), (1, 4, 3)
+    if xzero_point_zero == True:
+        x_zero_point = 0
+    else:
+        x_zero_point = -1
+
+    if yzero_point_zero == True:
+        y_zero_point = 0
+    else:
+        y_zero_point = -1
```
Comment (masahi) [inline tests/python/relay/test_op_qnn_batch_matmul.py:85]: Should test on more non-zero zero points.

**Comment by masahi** [comment]
@elvin-n Thanks. This is going to be useful for quantized transformers.

cc @anijain2305

File: tests/python/relay/test_op_qnn_batch_matmul.py (hunk 2 above)
Comment (elvin-n) [reply tests/python/relay/test_op_qnn_batch_matmul.py:85]: @masahi just curious if changing -1 to other value would be enough or you propose to add other tests. 
Currently there are 5 test cases:
x zp =0, y zp = 0
x zp = -1, y zp = 0
x zp = 0, y zp = -1
x zp =-1, y zp =-1
that covers all flows in `QnnBatchMatmulCanonicalize`

File: python/tvm/relay/qnn/op/qnn.py (hunk 1 above)
Comment (elvin-n) [reply python/tvm/relay/qnn/op/qnn.py:696]: fixed

File: tests/python/relay/test_op_qnn_batch_matmul.py (hunk 2 above)
Comment (masahi) [reply tests/python/relay/test_op_qnn_batch_matmul.py:85]: I think we should test on larger non-zero zero points, to verify the accuracy.

Comment (elvin-n) [reply tests/python/relay/test_op_qnn_batch_matmul.py]: modified zp to 123

**Review by masahi — APPROVED** [review]

**Comment by masahi** [comment]
Thanks @elvin-n @jwfromm

**Pull request merged**